src/box/alter.cc

```diff
@@ -4418,9 +4418,25 @@ on_replace_dd_schema(struct trigger * /* trigger */, void *event)
 
 /** Unregister the replica affected by the change. */
 static int
-on_replace_cluster_clear_id(struct trigger *trigger, void * /* event */)
-{
-	replica_clear_id((struct replica *)trigger->data);
+on_replace_cluster_clear_id(struct trigger *trigger, void *event)
+{
+	struct replica *replica = (struct replica *)trigger->data;
+	struct txn_stmt *stmt = (struct txn_stmt *)event;
+	if (replica->id == instance_id) {
+		assert(stmt->row->replica_id != 0 ||
+		       recovery_state != FINISHED_RECOVERY);
+		struct replica *origin = replica_by_id(stmt->row->replica_id);
+		if (recovery_state == FINISHED_RECOVERY && origin != NULL) {
+			assert(origin->applier != NULL &&
+			       origin->applier->fiber != NULL);
```

Comment:
Nit: we already assert, that `fiber` is not `NULL` in `applier_kill`. Seems redundant

Reply:
Fixed.